packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingAnnotation.spec.js

```diff
@@ -1,19 +1,187 @@
-  value: [ point.id ]
-}
 
-describe('Model > DrawingAnnotation', function () {
+  const drawingAnnotationSnapshot = {
+    id: 'drawing1',
+    task: 'T0',
+    taskType: 'drawing',
+    value: [ point.id ]
+  }
+
   it('should exist', function () {
     const drawingAnnotation = DrawingAnnotation.create(drawingAnnotationSnapshot)
     expect(drawingAnnotation).to.exist()
     expect(drawingAnnotation).to.be.an('object')
   })
+
+  describe('annotation snapshots', function () {
+    describe('without subtasks', function () {
+      let snapshot
+
+      before(function () {
+        const drawingTask = DrawingTask.create({
+          instruction: 'draw things!',
+          taskKey: 'T0',
+          tools: [
+            {
+              type: 'point'
+            }
+          ],
+          type: 'drawing'
+        })
+        const pointTool = drawingTask.tools[0]
+        const point = pointTool.createMark({
+          x: 50,
+          y: 100
+        })
+        const annotation = DrawingAnnotation.create({
+          id: 'drawing1',
+          task: 'T0',
+          taskType: 'drawing',
+          value: [ point ]
+        })
+        const mockStore = types.model('MockStore', {
+          drawingTask: DrawingTask,
+          annotation: DrawingAnnotation
+        })
+        .create({
+          drawingTask,
+          annotation
+        })
+        snapshot = annotation.toSnapshot()
+      })
+
+      it('should contain exactly one annotation', function () {
+        expect(snapshot).to.have.lengthOf(1)
+      })
+
+      describe('the annotation snapshot', function () {
+        it('should contain a task key', function () {
+          const [ annotation ] = snapshot
+          expect(annotation.task).to.equal('T0')
+        })
+
+        it('should contain a taskType', function () {
+          const [ annotation ] = snapshot
+          expect(annotation.taskType).to.equal('drawing')
+        })
+
+        it('should include snapshots of the drawn marks', function () {
+          const [ annotation ] = snapshot
+          const pointSnapshot = {
+            details: [],
+            frame: 0,
+            toolIndex: 0,
+            toolType: 'point',
+            x: 50,
+            y: 100
+          }
+          expect(annotation.value).to.deep.equal([ pointSnapshot ])
+        })
+      })
+    })
+
+    describe('with subtasks', function () {
+      let snapshot
+
+      before(function () {
+        const drawingTask = DrawingTask.create({
+          instruction: 'draw things!',
+          taskKey: 'T0',
+          tools: [
+            {
+              type: 'point',
+              details: [{
+                type: 'single',
+                answers: ['yes', 'no'],
+                question: 'yes or no?'
+              }]
+            }
+          ],
+          type: 'drawing'
+        })
+        const annotation = DrawingAnnotation.create(drawingAnnotationSnapshot)
+        const mockStore = types.model('MockStore', {
```

Comment:
Same comment about mocking parent stores.

Reply:
A very good rule to follow when testing is only write as much code as you need to get the job done. I’m very strongly against adding unnecessary complication to the tests. 

Here, a drawing annotation is a collection of references to a drawing task, so the test mocks a task and an annotation. There’s no dependence on any other models so they aren’t included.

Reply:
Again, this is testing an annotation in isolation.

Reply:
Ok, we disagree then.